R/simpleCache.R

```diff
@@ -93,183 +100,197 @@ simpleCache = function(cacheName, instruction=NULL, buildEnvir=NULL, reload=FALS
-		message("::Object exists (in ", cacheWhere, ")::\t", cacheName);
-		ret = get(cacheName, pos=get(cacheWhere));
+		message("::Object exists (in ", cacheWhere, ")::\t", cacheName)
+		#return(get(cacheName))
+		#return()
+		ret = get(cacheName, pos = get(cacheWhere))
 	} else if (file.exists(lockFile) & !ignoreLock) {
 		message("::Cache processing (lock file exists)::\t", lockFile)
-		# Check for slurm log...
-		if (!is.null(slurmParams)) { slurmLog = paste0(slurmParams$hpcFolder, "/", cacheName, ".log"); message(slurmLog); tail(readLines(slurmLog), 10) }
+		#check for slurm log...
+
+		if (!is.null(batchRegistry)) {
+			# Grabbing log from batchtools
+			# 1 is the job id.
+			message(paste(batchtools::getLog(1, reg=registry), collapse="\n"))
+		}
+		if (!is.null(pepSettings)) { 
+			# TODO: retrieve log
+			slurmLog = file.path(slurmParams$hpcFolder, paste0(cacheName, ".log"))
+			message(slurmLog)
+			utils::tail(readLines(slurmLog), 10) 
+		}
+
+		return()
 
-		return ();
 	} else if(file.exists(cacheFile) & !recreate & !noload) {
-		message("::Loading cache::\t", cacheFile);
-		load(cacheFile);
+		message("::Loading cache::\t", cacheFile)
+		load(cacheFile)
 	} else if(file.exists(cacheFile) & !recreate) {
-		message("::Cache exists (no load)::\t", cacheFile);
-		return (NULL);
+		message("::Cache exists (no load)::\t", cacheFile)
+		return(NULL)
 	} else {
-		message("::Creating cache::\t", cacheFile);
+		message("::Creating cache::\t", cacheFile)
 
 		tryCatch( { # Intercept any errors with creating this cache.
 
 		if(is.null(instruction)) {
 				if (is.null(buildDir)) {
-					stop("::Error::\tIf you do not provide an instruction argument, you must set global option RBUILD.DIR with setCacheBuildDir, or specify a buildDir parameter directly to simpleCache().");
+					stop(strwrap("::Error::\tIf you do not provide an
+					instruction argument, you must set global option RBUILD.DIR
+					with setCacheBuildDir, or specify a buildDir parameter
+					directly to simpleCache()."))
 				}
-				RBuildFile = paste0(buildDir, cacheName, ".R");
+				RBuildFile = file.path(buildDir, paste0(cacheName, ".R"))
+
 				if (!file.exists(RBuildFile)) {
-					stop("::Error::\tNo instruction or RBuild file provided.");
+					stop("::Error::\tNo instruction or RBuild file provided.")
 				}
-				if (timer) { tic(); }
-				source(paste0(buildDir, cacheName, ".R"), local=FALSE);
-				if (timer) { toc(); }
-				ret = get(cacheName);
+
+				if (timer) { tic() }
+				source(file.path(buildDir, paste0(cacheName, ".R")), local=FALSE)
+				if (timer) { toc() }
+				ret = get(cacheName)
 		} else {
-			#"ret," for return, is the name the cacheName is stored under.
+			
 			if (is.null(buildEnvir)) {
-				if (timer) { tic(); }
-				if (is.null(slurmParams)) {
-					
+				if (timer) { tic() }
+				if ( ! is.null(batchRegistry) ) {
+					# Submit to cluster using batchtools 
+
+					 if (! requireNamespace("batchtools", quietly=TRUE)) {
```

Comment:
Question from the uninitiated here...what are the pros/cons of `quietly = TRUE` followed by an error vs. just `quietly = FALSE`?